### PR TITLE
Jest circleci

### PR DESCRIPTION
**Pull request by didierofrivia**
**What this PR does / why we need it**:

After upgrading webpack and babel, jest tests are  costly.

**Which issue(s) this PR fixes** 
fixes https://issues.jboss.org/browse/THREESCALE-2833

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -565,6 +565,7 @@ jobs:
       - *upload-coverage
 
   jest:
+    resource_class: medium+
```
Comment (josemigallas) [inline .circleci/config.yml]: Do we have budget approval for this?

Comment (didierofrivia) [reply .circleci/config.yml]: @thomasmaas @gsaslis what do you reckon?

Comment (thomasmaas) [reply .circleci/config.yml]: Let’s just do it, we can always undo it

Comment (josemigallas) [reply .circleci/config.yml]: Apparently a support ticket has to opened, in order for this to work.

Comment (josemigallas) [reply .circleci/config.yml]: It LGTM to merge but let's be sure it won't be for nothing.

Comment (gsaslis) [reply .circleci/config.yml]: yup just go ahead. we already have the option to use different resource classes enabled. (and we do use it already in porta ci). 

btw, this won't necessarily cost _that much more_ because if jest takes advantage of the extra cores and the test finish "faster" (and billing is per minute), it won't be "double" the cost. 

tl;dr go go go go go go go go go

**Comment by Martouta** [comment]
Please @didierofrivia rebase from master 😄

**Comment by josemigallas** [comment]
I rebased onto master, could you guys approve again? @gsaslis @hallelujah
